src/mlpack/core/math/digamma.hpp

```diff
@@ -0,0 +1,266 @@
+/**
+ * @file core/math/digamma.hpp
+ * @author Gopi Tatiraju 
+ *
+ * Some parts of the implementation are inspired from boost.
+ * Boost Software License, Version 1.0. (See accompanying file
+ * LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+ * 
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_MATH_DIGAMMA_HPP
+#define MLPACK_CORE_MATH_DIGAMMA_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace math {
+
+
+/**
+ * This function evaluates the polynomial based on the
+ * constants passed to the function.
+ *
+ * This function evaluates the polynomial when x > 10.
+ *
+ * @param a Array of constants.
+ * @param x Input for which digamma will be calculated.
+ */
+template<typename T>
+T EvaluatePolyLarge(const T* a, const T& x)
+{
```

Comment:
@heisenbuug I think we should have `N==7` here, the same way you have done it to other functions right? otherwise, I imagine that this function can be executed if `N==6` and it will segfault in this case.

Reply:
No, this function is only being used in one place.
So, this will be only called once, so no issues.
But if you you suggest that it's safe to add `N==7`, I can do that.

Reply:
I think it is safer to put `N==7`, otherwise it might cause errors